tests/conftest.py

```diff
@@ -128,8 +142,14 @@ def _inner(srcpath, args=None):
             srcpath:
             args: Dictionary of arguments.
         """
+        if not args or not args.get('workflow_name', ''):
+            id_ = test_workflow_name()
+            args = {'workflow_name': id_}
+
+        args.update({'no_run_name': True})
         options = Options(install_gop(), args)()
         output = SimpleNamespace()
+        output.id = args['workflow_name']
```

Comment:
FYI, this is going to conflict when merged into master because all the `output.ret`, `output.exc`, `output.out`, etc stuff was stripped in favour of native pytest functionalities.

I think what you've added here mirrors the behaviour on master so should be easy to reconcile.

https://github.com/cylc/cylc-rose/blob/ed485a84b0f375d4dee2b7df7676d92712f0f905/tests/conftest.py#L175-L205